sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsScalerProvider.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Azure.Messaging.EventHubs.Primitives;
+using Microsoft.Azure.WebJobs.EventHubs.Processor;
+using Microsoft.Azure.WebJobs.Extensions.EventHubs.Listeners;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.EventHubs.Listeners
+{
+    internal class EventHubsScalerProvider : IScaleMonitorProvider, ITargetScalerProvider
+    {
+        private readonly IScaleMonitor _scaleMonitor;
+        private readonly ITargetScaler _targetScaler;
+
+        public EventHubsScalerProvider(IServiceProvider serviceProvider, TriggerMetadata triggerMetadata)
+        {
+            AzureComponentFactory azureComponentFactory;
+            if ((triggerMetadata.Properties != null) && (triggerMetadata.Properties.TryGetValue(nameof(AzureComponentFactory), out object value)))
+            {
+                azureComponentFactory = value as AzureComponentFactory;
+            }
+            else
+            {
+                azureComponentFactory = serviceProvider.GetService<AzureComponentFactory>();
+            }
+
+            var configuration = serviceProvider.GetService<IConfiguration>();
+            var hostComponentFactory = serviceProvider.GetService<AzureComponentFactory>();
+            var logForwarder = serviceProvider.GetService<AzureEventSourceLogForwarder>();
+            var options = serviceProvider.GetService<IOptions<EventHubOptions>>();
+            var loggerFactory = serviceProvider.GetService<ILoggerFactory>();
+            var checkpointClientProvider = serviceProvider.GetService<CheckpointClientProvider>();
+            var nameResolver = serviceProvider.GetService<INameResolver>();
+            var evetnHubMetadata = JsonConvert.DeserializeObject<EventHubMetadata>(triggerMetadata.Metadata.ToString());
+            var factory = new EventHubClientFactory(configuration, hostComponentFactory, options, nameResolver, logForwarder, checkpointClientProvider);
+            evetnHubMetadata.ResolveProperties(serviceProvider.GetService<INameResolver>());
+            var eventHubConsumerClient = factory.GetEventHubConsumerClient(evetnHubMetadata.EventHubName, evetnHubMetadata.Connection, evetnHubMetadata.ConsumerGroup);
+            var checkpointStore = new BlobCheckpointStoreInternal(
```

Comment:
How are we providing the storage client to use in the blob checkpoint store? We need to make sure this integrates iwth MSI, as the scale controller connects to the customer's storage account where the checkpoint logs are persisted

Reply:
I changed how `CheckpointClientProvider` is created - it should support MSI now.